pkg/cmd/roachtest/sequelize.go

```diff
@@ -0,0 +1,154 @@
+var supportedSequelizeRelease = "v6.0.0-alpha.0"
+
+// This test runs sequelize's full test suite against a single cockroach node.
+
+func registerSequelize(r *testRegistry) {
+	runSequelize := func(
+		ctx context.Context,
+		t *test,
+		c *cluster,
+	) {
+		if c.isLocal() {
+			t.Fatal("cannot be run in local mode")
+		}
+		node := c.Node(1)
+		t.Status("setting up cockroach")
+		c.Put(ctx, cockroach, "./cockroach", c.All())
+		if err := c.PutLibraries(ctx, "./lib"); err != nil {
+			t.Fatal(err)
+		}
+		c.Start(ctx, t, c.All())
+
+		version, err := fetchCockroachVersion(ctx, c, node[0])
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if err := alterZoneConfigAndClusterSettings(ctx, version, c, node[0]); err != nil {
+			t.Fatal(err)
+		}
+
+		t.Status("create database used by tests")
+		db, err := c.ConnE(ctx, node[0])
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer db.Close()
+
+		if _, err := db.ExecContext(
+			ctx,
+			`CREATE DATABASE sequelize_test`,
+		); err != nil {
+			t.Fatal(err)
+		}
+
+		t.Status("cloning Sequelize and installing prerequisites")
+		latestTag, err := repeatGetLatestTag(ctx, c, "cockroachdb", "sequelize-cockroachdb", sequelizeReleaseTagRegex)
+		if err != nil {
+			t.Fatal(err)
+		}
+		c.l.Printf("Latest Sequelize release is %s.", latestTag)
+		c.l.Printf("Supported Sequelize release is %s.", supportedSequelizeRelease)
+
+		if err := repeatRunE(
+			ctx, c, node, "update apt-get", `sudo apt-get -qq update`,
+		); err != nil {
+			t.Fatal(err)
+		}
+
+		if err := repeatRunE(
+			ctx,
+			c,
+			node,
+			"install dependencies",
+			`sudo apt-get -qq install make python3 libpq-dev python-dev gcc g++ `+
+				`python-software-properties build-essential`,
+		); err != nil {
+			t.Fatal(err)
+		}
+
+		if err := repeatRunE(
+			ctx,
+			c,
+			node,
+			"add nodesource repository",
+			`curl -sL https://deb.nodesource.com/setup_12.x | sudo -E bash -`,
+		); err != nil {
+			t.Fatal(err)
+		}
+
+		if err := repeatRunE(
+			ctx, c, node, "install nodejs and npm", `sudo apt-get -qq install nodejs`,
+		); err != nil {
+			t.Fatal(err)
+		}
+
+		if err := repeatRunE(
+			ctx, c, node, "update npm", `sudo npm i -g npm`,
+		); err != nil {
+			t.Fatal(err)
+		}
+
+		if err := repeatRunE(
+			ctx, c, node, "remove old sequelize", `sudo rm -rf /mnt/data1/sequelize`,
+		); err != nil {
+			t.Fatal(err)
+		}
+
+		if err := repeatGitCloneE(
+			ctx,
+			t.l,
+			c,
+			"https://github.com/cockroachdb/sequelize-cockroachdb.git",
+			"/mnt/data1/sequelize",
+			supportedSequelizeRelease,
+			node,
+		); err != nil {
+			t.Fatal(err)
+		}
+
+		if err := repeatRunE(
+			ctx, c, node, "install dependencies", `cd /mnt/data1/sequelize && sudo npm i`,
+		); err != nil {
+			t.Fatal(err)
+		}
+
+		t.Status("running Sequelize test suite")
+		rawResults, err := c.RunWithBuffer(ctx, t.l, node,
+			`cd /mnt/data1/sequelize/ && npm test`,
+		)
+		rawResultsStr := string(rawResults)
```

Comment:
i think we'd want to have the blocklists/ignorelists in place too. otherwise we wouldn't know if it fails

or if no failures are expected, then the roachtest should just automatically fail if any tests fail and point people at the logs

Reply:
Yeah I'm gonna add that after, for some reason I had this test working but now I'm running into a certs issue :/

Reply:
> i think we'd want to have the blocklists/ignorelists in place too. otherwise we wouldn't know if it fails

isn't that what the `err != nil` check does below? `RunWithBuffer` returns an error if exit error code != 0

Reply:
oh whoops yes!